### ISSUE TITLE
FileNotFoundError in hapmultisequencer after file was created

### ISSUE THREAD

**Issue by stscijgbot-hstdp** [spacetelescope/drizzlepac#1055]
_Issue [HLA-580](https://jira.stsci.edu/browse/HLA-580) was created on JIRA by [Robert Swaters](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rswaters):_

```
2021174215014 INFO src=drizzlepac.haputils.align_utils- Adding hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_flc.fits to HAPImage list
2021174215146 INFO src=drizzlepac.align- RESULTS FOR hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_flc.fits Chip 1: FIT_RMS = 9.600207407735544 mas, TOTAL_RMS = 9.600207407735544 mas, NUM =  26
2021174215146 INFO src=drizzlepac.align- RESULTS FOR hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_flc.fits Chip 2: FIT_RMS = 9.600207407735544 mas, TOTAL_RMS = 9.600207407735544 mas, NUM =  26
    Blot: creating blotted image:  hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_flc.fits[sci,1]
    Blot: creating blotted image:  hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_flc.fits[sci,2]
FileNotFoundError: [Errno 2] No such file or directory: 'hst_skycell-p2381x06y09_wfc3_uvis_f547m_fine-all-all_icnk20ay_fl'
```
Given the error message, maybe the filename is incorrectly cut short?

Full log is attached.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Warren Hack](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hack) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535397#comment-535397):_

This problem stems from the code relying on the D001DATA keywords in 'processing_utils.find_footprint()', which documents the names of all the inputs that contributed to a drizzle product.  Unfortunately, the MVM filenames are so long that they get truncated by the FITS standard (<70 chararacters for a single keyword value).  

The only viable solution would be to add code in that function to auto-complete the filename.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Warren Hack](https://jira.stsci.edu/secure/ViewProfile.jspa?name=hack) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535398#comment-535398):_

This is listed as a 'BLOCKER' since it prevents the MVM products from being updated to be compatible with CAOM.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535403#comment-535403):_

Actually, the FITS Standard now has the concept of long strings which are greater than 68 characters, excluding the single quotes around the string.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535649#comment-535649):_

Looking ahead...where can I get the data which caused this issue in the first place?

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Robert Swaters](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rswaters) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535667#comment-535667):_

Here's the poller file:
```
icnk20a8q_flc.fits,13773,CNK,20,515.0,F657N,UVIS,skycell-p2381x06y09,NEW,/ifs/archive/dev/hst/info/rswaters/test/test_rswaters_20210623_01/regr_08130/pipe/science/hap/icnk20a8q_flc.fits
icnk20adq_flc.fits,13773,CNK,20,515.0,F657N,UVIS,skycell-p2381x06y09,NEW,/ifs/archive/dev/hst/info/rswaters/test/test_rswaters_20210623_01/regr_08130/pipe/science/hap/icnk20adq_flc.fits
icnk20afq_flc.fits,13773,CNK,20,515.0,F657N,UVIS,skycell-p2381x06y09,NEW,/ifs/archive/dev/hst/info/rswaters/test/test_rswaters_20210623_01/regr_08130/pipe/science/hap/icnk20afq_flc.fits
icnk20awq_flc.fits,13773,CNK,20,354.0,F547M,UVIS,skycell-p2381x06y09,NEW,/ifs/archive/dev/hst/info/rswaters/test/test_rswaters_20210623_01/regr_08130/pipe/science/hap/icnk20awq_flc.fits
icnk20ayq_flc.fits,13773,CNK,20,200.0,F547M,UVIS,skycell-p2381x06y09,NEW,/ifs/archive/dev/hst/info/rswaters/test/test_rswaters_20210623_01/regr_08130/pipe/science/hap/icnk20ayq_flc.fits
```
Please let me know if you can't access the files listed in the poller file.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535713#comment-535713):_

This bug has been addressed in Git PRs #&#x2060;1057 and #&#x2060;1058. The PRs have gone out for code review.  Please note that now the "long string" values can be written to their keywords, their associated comments may be abbreviated or missing.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Michele De La Pena](https://jira.stsci.edu/secure/ViewProfile.jspa?name=mdelapena) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535755#comment-535755):_

The fix for this ticket has been merged onto drizzlepac master (Git PRs #&#x2060;1057 and #&#x2060;1058).

I will note now that the long string values are not truncated, you may see more FITS cards in some headers as the keyword values can be written out fully (e.g., S_REGION can be many cards long).  One downside I noticed is if the keyword value really just needs one card or less to be written out fully, it is possible for the commentary portion of the card to be abbreviated or missing entirely.

Please feel free to pull the latest code from drizzlepac master for your testing.

**Comment by stscijgbot-hstdp** [comment]
_Comment by [Robert Swaters](https://jira.stsci.edu/secure/ViewProfile.jspa?name=rswaters) on [JIRA](https://jira.stsci.edu/browse/HLA-580?focusedCommentId=535815#comment-535815):_

Tests with the latest version confirm that the FileNotFoundError resported above has been resolved. Unfortunately, a new problems seems to have popped up, subject of HLA-582.

**Comment by stsci-hack** [comment]
Closed by #1057 and #1058
